go/tools/builders/link.go

```diff
@@ -110,33 +109,34 @@ func link(args []string) error {
 	parseXdef := func(xdef string) (pkg, name, value string, err error) {
 		eq := strings.IndexByte(xdef, '=')
 		if eq < 0 {
-			return "", "", "", fmt.Errorf("-X or -Xstamp flag does not contain '=': %s", xdef)
+			return "", "", "", fmt.Errorf("-X flag does not contain '=': %s", xdef)
 		}
 		dot := strings.LastIndexByte(xdef[:eq], '.')
 		if dot < 0 {
-			return "", "", "", fmt.Errorf("-X or -Xstamp flag does not contain '.': %s", xdef)
+			return "", "", "", fmt.Errorf("-X flag does not contain '.': %s", xdef)
 		}
 		pkg, name, value = xdef[:dot], xdef[dot+1:eq], xdef[eq+1:]
 		if pkg == *packagePath {
 			pkg = "main"
 		}
 		return pkg, name, value, nil
 	}
-	for _, xdef := range xstamps {
-		pkg, name, key, err := parseXdef(xdef)
-		if err != nil {
-			return err
-		}
-		if value, ok := stampMap[key]; ok {
-			goargs = append(goargs, "-X", fmt.Sprintf("%s.%s=%s", pkg, name, value))
-		}
-	}
 	for _, xdef := range xdefs {
 		pkg, name, value, err := parseXdef(xdef)
 		if err != nil {
 			return err
 		}
-		goargs = append(goargs, "-X", fmt.Sprintf("%s.%s=%s", pkg, name, value))
+		var missingKey bool
+		value = regexp.MustCompile(`\{.+?\}`).ReplaceAllStringFunc(value, func(key string) string {
```

Comment:
I think that `\{.+?\}` would more commonly be written as `\{.*\}`. If you make it that way I'm happy to merge.

Reply:
As I said in the previous comment, `.+?` is not equivalent to `.*` here. The `?` right after `*` doesn't  represent "optional". It actually enable [lazy matching](https://en.wikipedia.org/wiki/Regular_expression#Lazy_matching) here.

Reply:
Ah, apologies around that. As is usually the case when I interact with regex I learned something. Thanks for teaching me!